Added download command

Dym03 opened the pull request:
Added download feature to cargo kelvin
Parameters are assignment_id, login, download_type, output_name
With only assignment_id and login being mandatory, download_type is set to homework by default

Structure of the command with example 
```
cargo kelvin download <assignment_id> --download_type <download_type>
cargo kelvin download 9264 --login DYM0018 --download_type homework
```
This would download homework from the assignment with the specified id and extract it into a directory

Dym03 committed:
Added download command and updated documentation

Dym03 commented:
Made changes proposed by a colleague (Patrick) to get the login from the kelvin info api, instead of taking it as a param :)

Kobzol commented on src/main.rs:
```diff
@@ -49,6 +52,45 @@ struct SubmitArgs {
+    Exercise,
+}
+
+impl ToString for DownloadOption {
```
It's more common to implement `Display`.

Dym03 replied:
As it was never used, i removed it completely.

Kobzol commented on src/main.rs:
```diff
@@ -117,6 +177,104 @@ fn main() -> anyhow::Result<()> {
+        log::debug!(
+            "Response content: {}",
+            res.text().context("getting content of HTTP response")?
+        );
```
Should return `Err`?

Kobzol commented on src/main.rs:
```diff
@@ -117,6 +177,104 @@ fn main() -> anyhow::Result<()> {
+                        .send()
+                        .expect(&format!("Dowloading file {} failed", file));
+
+                    let mut out = File::create("zip.zip").expect("failed to create file");
```
```rust
let response: Vec<u8> = zip_resp.bytes()?.to_vec();
extract_zip(Cursor::new(response), &output_name);
```
You don't need to write to disk :)

Kobzol commented on src/main.rs:
```diff
@@ -210,3 +368,56 @@ fn write_file_to_zip<W: Write + Seek>(
+                set_permissions(&outpath, Permissions::from_mode(mode)).unwrap();
+            }
+        }
+    }
```
```suggestion
    archive.extract(".").unwrap();
```

Kobzol reviewed (commented):
Great work, thank you! Left some comments :)

Dym03 commented:
Made changes that were suggested. Added lecture download_type for code from lecture. Removed the initially proposed -o flag for the output directory, as it would made the extraction bulkier.

Kobzol reviewed (approved):
Looks great, thank you! :)
